### ISSUE TITLE
PostgreSQL \crosstabview breaks formatting

### ISSUE THREAD

**Issue by pgr0ss** [sqlfluff/sqlfluff#7815]
### Search before asking

- [x] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

I have a SQL file which uses `\crosstabview` (https://wiki.postgresql.org/wiki/Crosstabview) and sqlfluff is unable to parse/format it.

Here is an example `crosstab.sql` file:
```
select
    a,
    b,
    c
from mytable
order by 1
\crosstabview

select
    a,
    b,
    c
from mytable
order by 1
\crosstabview
```

And trying to format this with v4.1.0:

```bash
> uvx sqlfluff@4.1.0 format --dialect postgres crosstab.sql
==== finding fixable violations ====
  [1 templating/parsing errors found]
  == [crosstab.sql] ==
  L:   9 | P:   1 |  PRS | Line 9, Position 1: Found unparsable section: 'select\n
                       | a,\n    b,\n    c\nfrom mytable\n...'
==== no fixable linting violations found ====
All Finished 📜 🎉!
```

### Expected Behaviour

I expected the file to parse and format. It seems like everything after the `\crosstabview` will fail parsing. If I remove the entire second query, the file actually seems to parse.

### Observed Behaviour

See above

### How to reproduce

See above

### Dialect

postgres

### Version

```bash
> uvx sqlfluff@4.1.0 --version
sqlfluff, version 4.1.0

> uvx python --version
Python 3.12.4
```

### Configuration

None

### Are you willing to work on and submit a PR to address the issue?

- [x] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by pgr0ss** [comment]
I don't think this is fully fixed. Here's a more complicated example:

`crosstab2.sql`:
```sql
select
    a,
    b,
    c
from mytable
order by 1
\crosstabview

with entries as (select 1, 2, 3)
select
    a,
    b,
    c
from entries
order by 1
\crosstabview
```

```bash
> uvx sqlfluff@4.2.1 format --dialect postgres crosstab2.sql
==== finding fixable violations ====
== [crosstab2.sql] FAIL
  [1 templating/parsing errors found]
  == [crosstab2.sql] ==
  L:  20 | P:   1 |  PRS | Line 20, Position 1: Found unparsable section:
                       | '\\crosstabview'
==== no fixable linting violations found ====
All Finished 📜 🎉!
  [6 unfixable linting violations found]
```
